### PR TITLE
File share Mount target resource rename

### PR DESCRIPTION
**Pull request by deepaksibm**
Deprecated ibm_is_share_target resource and datasources
Added ibm_is_share_mount_target

### CODE REVIEW

**Comment by deepaksibm** [comment]
```

$ tfpt TestAccIbmIsShareMountTargetBasic
--- PASS: TestAccIbmIsShareMountTargetBasic (113.26s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc
```

**Comment by deepaksibm** [comment]
```

$ tfpt TestAccIbmIsShareMountTarget
--- PASS: TestAccIbmIsShareMountTargetBasic (108.46s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc	220.087s
```

File: ibm/service/vpc/resource_ibm_is_share_mount_target.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+	if err != nil {
+		log.Printf("[DEBUG] CreateShareMountTargetWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+	_, err = WaitForTargetAvailable(context, vpcClient, *createShareMountTargetOptions.ShareID, *shareTarget.ID, d, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", *createShareMountTargetOptions.ShareID, *shareTarget.ID))
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_share_mount_target.go:129]: Can we setID before waitingfor available. If soemtimes waiting intermittely the state file will hold ID and helps in future run

File: ibm/service/vpc/resource_ibm_is_share_mount_target.go [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+	shareTarget, response, err := vpcClient.GetShareMountTargetWithContext(context, getShareMountTargetOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetShareMountTargetWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_share_mount_target.go:159]: Can we check shareTarget != nil and then access *shareTarget.X for eliminating nil pointer exception .
or Add in above 

if err != nil  || shareTarget = nil{

File: ibm/service/vpc/resource_ibm_is_share_mount_target.go (hunk 1 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_share_mount_target.go]: moved above wait logic call

File: ibm/service/vpc/resource_ibm_is_share_mount_target.go (hunk 2 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_share_mount_target.go:159]: added shareTarget == nil check

**Comment by deepaksibm** [comment]
adding a fix. moving to draft till then

**Comment by deepaksibm** [comment]
Fix added and tested. PR is ready to review and merge

**Pull request merged**